api/v1alpha1/module_types.go

```diff
@@ -58,6 +58,10 @@ type Build struct {
 
 	// Push contains settings determining how to push a built DriverContainer image.
 	Push PushOptions `json:"push"`
+
+	// Secrets is an optional list of secrets to be made available to the build system.
+	// +optional
+	Secrets []v1.LocalObjectReference `json:"secrets"`
```

Comment:
This limits the object reference to the same namespace, but what does it mean for non namespaced resources? I think the entity creating the `Module` CR will own the secret and store them in its own namespace.

Reply:
Created #43 to transition `Module` to a namespaced resource.